### PR TITLE
WIP: Export body files / body / all

### PR DESCRIPTION
**Pull request by ns3777k**
Since bodyFiles get loaded into `body` on import, `body` is returned filled on export along with `bodyFile`. When both fields provided a warning is printed and hoverfly will use only body.

One way to get around this behavior is to add some flag to `/api/v2/simulation` that defines how to export the simulation. I've added `bodyMode` that can configure this based on the value:

1. `body-only`: export only `body`, emptying all bodyFile fields (which is default)
2. `prefer-bodyfile`: export both fields, but set `body` to an empty string when `bodyFile` is not empty (maybe this should be the defaut?)
3. `all`: export both `body` and `bodyFile` as is (current behavior)

It's just a quick working example to show a concept. It builds but tests don't pass for now. Refactoring required. Don't merge :warning:

### CODE REVIEW

**Comment by ns3777k** [comment]
@tommysitu how do you think? would this approach that solve the exporting issue?

**Comment by tommysitu** [comment]
Hey @ns3777k thank you for suggesting a solution. How about doing something simple first: 

if the response has both `body` and `bodyFile`, exporting should create or update the body file with the content from `body`, and then clearing the `body` field. This is useful if the user has run some capture, and doing an export can keep body files up-to-date 

This can be expanded to support another feature is to allow users doing capture, and exporting a simulation with body stores in separate files.

So importing/ exporting simulation with body files workflow will be complete.

**Comment by ns3777k** [comment]
@tommysitu Sounds cool. As far as I understand, `hoverctl export` should create / update bodyFiles locally, right? What if bodyFile is an absolute path, where this file should be saved?

**Comment by tommysitu** [comment]
yeah, that's what hoverctl does, it calls the get simulation API, and save it to a file.

**Comment by ns3777k** [comment]
Yes, but if we run hoverctl export against a remote instance and get back an absolute path in bodyFile, we should not save it locally in the given absolute path for obvious security reasons, so the question is how to handle absolute paths when exporting?)

**Comment by tommysitu** [comment]
Agreed. I feel a bit wary about allow hoverctl to write to any absolute path autonomously on export.  We should enforce `bodyFile` path to be relative only. What do you think?

**Comment by ns3777k** [comment]
Actually that makes sense since we allow to set the path against the relative bodyFiles get resolved.

**Comment by ns3777k** [comment]
I've force pushed the branch. some tests still fail, I'll make em work later. Could you take a look please?

**Comment by tommysitu** [comment]
sure @ns3777k, will get back to you soon.

File: hoverctl/wrapper/simulation.go [hunk 1]
```diff
@@ -38,14 +37,27 @@ func ExportSimulation(target configuration.Target, urlPattern string) ([]byte, e
...
 
-	var jsonBytes bytes.Buffer
-	err = json.Indent(&jsonBytes, body, "", "\t")
-	if err != nil {
+	var view v2.SimulationViewV6
+	if err := json.Unmarshal(body, &view); err != nil {
 		log.Debug(err.Error())
-		return nil, errors.New("Could not export from Hoverfly")
+		return nil, err
+	}
```
Comment (tommysitu) [inline hoverctl/wrapper/simulation.go]: you can read response body into a struct in one go: 
```
err := json.NewDecoder(response.Body).Decode(&view)
```

Comment (ns3777k) [reply hoverctl/wrapper/simulation.go]: true

File: core/hoverfly_funcs.go [hunk 2]
```diff
@@ -127,7 +129,17 @@ func (hf *Hoverfly) readResponseBodyFiles(pairs []v2.RequestMatcherResponsePairV
 		}
 
 		if len(pair.Response.GetBody()) == 0 && len(pair.Response.GetBodyFile()) > 0 {
-			fileContents, err := ioutil.ReadFile(filepath.Join(hf.Cfg.ResponsesBodyFilesPath, pair.Response.GetBodyFile()))
+			bodyFile := pair.Response.GetBodyFile()
+			if filepath.IsAbs(bodyFile) {
+				err := errors.New(fmt.Sprintf(
```
[the remark this answers is not in the document]
Comment (ns3777k) [reply core/hoverfly_funcs.go]: true

**Comment by ns3777k** [comment]
I've fixed the current tests and the code your comments were referenced to. (I didn't write new tests and docs. I'll make a separate PR)

**Pull request merged**